docs/doc_index.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python3
+
+from pathlib import Path
+
+def _parse_entry(entry):
+    if isinstance(entry, str) and (entry.endswith('ipynb') or entry.endswith('rst')):
+        return example_entry(name=entry)
+    else:
+        return entry
+class Doc:
+    def __init__(self, title, options, entries):
+        self.title = title
+        self.options = options
+        self.entries = entries
+        self.entries = [_parse_entry(entry) for entry in entries]
+
+class ExampleEntry:
+    def __init__(self, name, operator_ref):
+        self.name = name
+        if operator_ref is not None:
+            if isinstance(operator_ref, list):
+                for elem in operator_ref:
+                    if not isinstance(elem, OpReference):
+                        raise TypeError(
+                            "Expected a single op_reference of a list of them to be provided")
+            elif not isinstance(operator_ref, OpReference):
+                raise TypeError("Expected a single op_reference of a list of them to be provided")
+        self.operator_ref = operator_ref
+
+    def __str__(self):
+        return self.name
+
+class OpReference:
+    def __init__(self, operator, docstring):
+        self.operator = operator
+        self.docstring = docstring
+
+
+def doc(title, options, entries):
```

Comment:
Here is the entry point for Python index file with the documentation. It's supposed to be used in the index.py files.